flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -159,11 +159,15 @@ void assignExclusiveSegments(List<MemorySegment> segments) {
 	 */
 	@VisibleForTesting
 	@Override
-	public void requestSubpartition(int subpartitionIndex) throws IOException, InterruptedException {
+	public void requestSubpartition(int subpartitionIndex) throws IOException {
 		if (partitionRequestClient == null) {
 			// Create a client and request the partition
-			partitionRequestClient = connectionManager
-				.createPartitionRequestClient(connectionId);
+			try {
+				partitionRequestClient = connectionManager.createPartitionRequestClient(connectionId);
+			} catch (Throwable t) {
```

Comment:
I think we should only catch `IOException` here. If a different exception is thrown, this indicates a serious Flink problem and we should best fail the `TaskExecutor` process.

Reply:
Only the `InterruptedException` might be thrown during `future.waitForChannel()`. If we only catch the `IOException`, then for other cases we could keep the previous behavior to throw the raw exception directly?